credoai/artifacts/data/tabular_data.py

```diff
@@ -72,25 +76,30 @@ def _process_y(self, y):
                 y = pd_type(y, name="target")
         return y
 
+    def _validate_X(self):
+        try:
+            pd.DataFrame(self.X)
+        except:
+            raise ValidationError("X must be able to be coerced into a dataframe")
+
     def _validate_y(self):
-        if self.X is not None:
-            """Basic validation for y"""
-            if len(self.X) != len(self.y):
-                raise ValidationError(
-                    "X and y are not the same length. "
-                    + f"X Length: {len(self.X)}, y Length: {len(self.y)}"
-                )
+        """Validation of Y inputs"""
+        if self.X is not None and (len(self.X) != len(self.y)):
+            raise ValidationError(
+                "X and y are not the same length. "
+                + f"X Length: {len(self.X)}, y Length: {len(self.y)}"
+            )
+
+    def _validate_processed_X(self):
+        """Validate processed X"""
+        if len(self.X.columns) != len(set(self.X.columns)):
+            raise ValidationError("X contains duplicate column names")
+        if len(self.X.index) != len(set(self.X.index)):
```

Comment:
It would make sense to also check that `sensitive_feature.columns` doesn't intersect with `X.columns` if both objects are specified.

Reply:
I don't think that's something we want to forbid. The sensitive features are used for sensitive feature things - they could exist in X as well (to be passed to the model)

Reply:
It won't do deceptively well... it will do really well and that is what we want. The features used for `sensitive_features` are distinct from the features used in X, which is the predictor features.